CMakeLists.txt

```diff
@@ -0,0 +1,12 @@
+cmake_minimum_required(VERSION 3.8)
```

Comment:
Why is the minimum 3.8?

Reply:
Requiring a higher version than needed is very annoying for people using the preinstalled cmake.

Not sure about 2.8.12 though; I thought INTERFACE was newer.

Reply:
To what preinstalled version are you referring? 3.5 from ubuntu 16.04 LTS? 

You are probably right that 2.8.12 didn't Interface libraries

Reply:
Is 3.5 an acceptable compromise then? 
I don't want any future contributor to have the feeling he can't use modern cmake functionalities just because this initial version didn't require anything more recent. Conversely, I don't want a user to rely on compilation with outdated cmake versions. Particularly considering that it will probably take quite some time until this reaches "regular" developers anyway.

Reply:
OK let's go with 3.5 and be done with it. And `BoostAssert` as the project name - so that the config file is called `BoostAssertConfig.cmake` when we get to it as this seems to be the convention nowadays.

Reply:
Ok, I made the changes.